src/utils/ethereumUtils.ts

```diff
@@ -364,6 +364,19 @@ const fetchTxWithAlwaysCache = async (address: EthereumAddress) => {
   return txTime;
 };
 
+export const fetchContractABI = async (address: EthereumAddress) => {
+  const url = `https://api.etherscan.io/api?module=contract&action=getabi&address=${address}&apikey=${ETHERSCAN_API_KEY}`;
```

Comment:
can we get the contract ABI via the provider instead? not sure but feels insecure to get it from etherscan